sno/dataset2.py

```diff
@@ -282,40 +282,51 @@ def iter_legend_blob_data(self):
                 blob.data,
             )
 
-    def import_iter_feature_blobs(self, resultset, source, replacing_dataset=None):
+    def import_iter_feature_blobs(
+        self, repo, resultset, source, replacing_dataset=None
+    ):
         schema = source.schema
-        if replacing_dataset is not None and replacing_dataset.schema != source.schema:
+        if replacing_dataset:
             # Optimisation: Try to avoid rewriting features for compatible schema changes.
-            change_types = replacing_dataset.schema.diff_type_counts(source.schema)
-            if not change_types["pk_updates"]:
-                # We can probably avoid rewriting all features.
-                for feature in resultset:
-                    try:
-                        pk_values = (feature[replacing_dataset.primary_key],)
-                        rel_path = self.encode_pks_to_path(pk_values, relative=True)
-                        existing_data = replacing_dataset.get_data_at(
-                            rel_path, as_memoryview=True
-                        )
-                    except KeyError:
-                        # this feature isn't in the dataset we're replacing
-                        yield self.encode_feature(feature, schema)
-                        continue
-
-                    existing_feature_raw_dict = replacing_dataset.get_raw_feature_dict(
-                        pk_values, data=existing_data
+            # this can take some time, but often results in much fewer git objects produced.
+            # For example, consider revision A with this feature:
```

Comment:
Try to avoid reusing the letters `a` and `b` more than once to mean unrelated things (even in different letter cases). You can use x and y, r1 and r2... anything else you like